enterprise/frontend/src/embedding-sdk/components/public/InteractiveDashboard/InteractiveDashboard.tsx

```diff
@@ -131,20 +136,49 @@ const InteractiveDashboardInner = ({
 };
 
 export const InteractiveDashboard = renderOnlyInSdkProvider(
-  ({ dashboardId, ...rest }: InteractiveDashboardProps) => {
-    const { id, isLoading } = useValidatedEntityId({
+  ({ dashboardId: initialDashboardId, ...rest }: InteractiveDashboardProps) => {
+    const { id: resolvedDashboardId, isLoading } = useValidatedEntityId({
       type: "dashboard",
-      id: dashboardId,
+      id: initialDashboardId,
     });
 
+    const errorPage = useSdkSelector(getErrorPage);
+    const dispatch = useSdkDispatch();
+    useEffect(() => {
+      if (resolvedDashboardId) {
+        dispatch(setErrorPage(null));
+      }
+    }, [dispatch, resolvedDashboardId]);
+
+    const { style, className } = rest;
     if (isLoading) {
-      return <SdkLoader />;
+      return (
+        <StyledPublicComponentWrapper className={className} style={style}>
+          <SdkLoader />
+        </StyledPublicComponentWrapper>
+      );
     }
 
-    if (!id) {
-      return <SdkError message="ID not found" />;
+    if (!resolvedDashboardId || errorPage?.status === 404) {
+      return (
+        <StyledPublicComponentWrapper className={className} style={style}>
+          <SdkError
+            message={jt`Dashboard ${(
+              <Code
+                bg="var(--mb-base-color-ocean-20)"
+                c="text-dark"
+                key="question-id"
+              >
+                {initialDashboardId}
+              </Code>
```

Comment:
Might be worth creating a component for this (considering we are using it in 3+ places)

Reply:
Fixed in 35ecb49a9fe582d2df14aca6890fedf33b52d430